### PR TITLE
not check failed status when doing live migration hook

### LINKED ISSUE #2652: Avoid more than 1 ongoing live migrations
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
It is possible to initiate two live migrations for the same VMI at the same time

**What you expected to happen**:
The system should have blocked me from initiating an additional live-migration until the first migration is completed (successfully or with failure)

**How to reproduce it (as minimally and precisely as possible)**:
1. Create VMI
2. Post one migration job for the VMI
3. Post another migration job for the VMI before the previous migration is completed

**Anything else we need to know?**:
The second migration attempt would probably be rejected by libvirt, but it would be better to block it earlier at the apiserver level

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:
the live migration hook check should only check completed status , not the failed statue which just indicate a error, sometimes a good live migratino only mark complete but no failed status
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Related to #2652 

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-api/webhooks/validating-webhook/admitters/migration-create-admitter.go [hunk 1]
```diff
@@ -88,8 +88,7 @@ func (admitter *MigrationCreateAdmitter) Admit(ar *v1beta1.AdmissionReview) *v1b
 	// are already in flight.
 	if vmi.Status.MigrationState != nil &&
 		string(vmi.Status.MigrationState.MigrationUID) != "" &&
-		!vmi.Status.MigrationState.Completed &&
-		!vmi.Status.MigrationState.Failed {
+		!vmi.Status.MigrationState.Completed {
```
Comment (slintes) [inline pkg/virt-api/webhooks/validating-webhook/admitters/migration-create-admitter.go]: hm, I think this is more tricky... when I see correct, an aborted migration has Completed = false and Failed = false, but starting a new migration would be fine imho. My next idea was to check if EndTimestamp > StartTimestamp, but on a quick look I think EndTimeStamp is not set for failed migrations... @vladikr, can you help please?

**Comment by ahadas** [comment]
I don't think it would be enough to resolve #2652: let's say we get two migration requests at the same time and no migration was executed before, both may pass this part of the validation webhook.

**Comment by jichenjc** [comment]
oh... that's a good suggestion, this seems need some lock introduced..

**Comment by ahadas** [comment]
> oh... that's a good suggestion, this seems need some lock introduced..

👍 
 
> but this should fix part of the issue?

Yeah, I believe it would fix part of the issue.
Another part I'm not so sure about is whether the following race is possible:
1. Request A gets to the apiserver
2. VirtualMachineInstanceMigration is added for request A
3. The apiserver returns positive response
4. Request B gets to the apiserver
5. VirtualMachineInstanceMigration is added for request B
6. The VMI is updated with the status of the migration initiated by request A

**Pull request closed without merging**